helper.js

```diff
@@ -1,57 +1,110 @@
-const fs = require('fs');
+const fs = require("fs");
+const path = require("path");
+const base64 = require("base-64");
+
+function getEnvPath(environment) {
+  const usersFilePath = path.join(__dirname, `users_${environment}.json`);
+  return usersFilePath;
+}
+
+function checkBasicAuthFromRequest(request) {
+  if (!request.headers || !request.headers.authorization) {
+    console.log("Request headers not passed");
+    return false;
+  }
+
+  const authHeader = request.headers.authorization;
+  if (!authHeader.startsWith("Basic ")) {
+    console.log("Basic authorization is missing");
+    return false;
+  }
+
+  const credentials = base64.decode(authHeader.substring(6));
+  const [username, password] = credentials.split(":");
+
+  const usersFilePath = path.join(__dirname, "users_basic_auth.json");
+  const users = JSON.parse(fs.readFileSync(usersFilePath, "utf-8"));
+
+  for (const user of users) {
+    if (user.username === username && user.password === password) {
+      console.log("User authorized to perform an action");
+      return true;
+    }
+  }
+  return false;
+}
 
 function login_change(username, loggedIn) {
-  const users = JSON.parse(fs.readFileSync('users_basic_auth.json', 'utf8'));
-  console.log(username)
-  const user = users.find(user => user.username === username);
+  const users = JSON.parse(fs.readFileSync("users_basic_auth.json", "utf8"));
+  console.log(username);
+  const user = users.find((user) => user.username === username);
 
   if (!user) {
-    throw new Error('Username not found');
+    throw new Error("Username not found");
   }
 
   // Update the loggedIn status for the user
   user.loggedIn = loggedIn;
 
-  fs.writeFileSync('users_basic_auth.json', JSON.stringify(users, null, 2));
+  fs.writeFileSync("users_basic_auth.json", JSON.stringify(users, null, 2));
 
   console.log(`Logged in status updated for user ${username}`);
 }
 
 // Read the users_basic_auth.json file and return its contents
 function getUsersCredentials() {
   try {
-    const usersCredentials = JSON.parse(fs.readFileSync('users_basic_auth.json', 'utf8'));
+    const usersCredentials = JSON.parse(
+      fs.readFileSync("users_basic_auth.json", "utf8"),
+    );
     return usersCredentials;
   } catch (error) {
-    console.error('Error reading users_basic_auth.json:', error.message);
+    console.error("Error reading users_basic_auth.json:", error.message);
     return [];
   }
 }
 
 // Function to write user data to the JSON file
-function writeUsersToFile(usersFilePath) {
+function writeUsersToFile(usersFilePath, environment) {
```

Comment:
either remove env or userFilePath. the environment parameter isn't used

Reply:
I have removed the env. It’s up thx ✋